src/terminal-screen.c

```diff
@@ -1684,33 +1690,57 @@ terminal_screen_button_press (GtkWidget      *widget,
 	TerminalScreen *screen = TERMINAL_SCREEN (widget);
 	gboolean (* button_press_event) (GtkWidget*, GdkEventButton*) =
 	    GTK_WIDGET_CLASS (terminal_screen_parent_class)->button_press_event;
-	char *matched_string;
-	int matched_flavor = 0;
+	char *hyperlink;
+	char *url;
+	int url_flavor = 0;
 	guint state;
 
 	state = event->state & gtk_accelerator_get_default_mod_mask ();
 
-	matched_string = terminal_screen_check_match (screen, (GdkEvent*)event, &matched_flavor);
+	hyperlink = terminal_screen_check_hyperlink (screen, (GdkEvent*)event);
+
+	if (hyperlink != NULL &&
+		(event->button == 1 || event->button == 2) &&
+		(state & GDK_CONTROL_MASK))
+	{
```

Comment:
I don't like the structure of this function. It conflates the `url` or `hyperlink` values with button press itself in the conditional statements, yet this is a button press callback function. I would suggest refactoring this to something like:
```c
	gboolean handled = FALSE;

	hyperlink = terminal_screen_check_hyperlink (screen, (GdkEvent*)event);
	url = terminal_screen_check_match (screen, (GdkEvent*)event, &url_flavor);

	if (event->button == 1 || event->button == 2) && (state & GDK_CONTROL_MASK))
	{
		if (hyperlink != NULL) // ...
		else if (url != NULL) // ...
	}

	if (event->button == 3 && (state & (GDK_SHIFT_MASK | GDK_CONTROL_MASK | GDK_MOD1_MASK)) == 0)
	{ /* ... popup stuff ... */ }

	g_free (hyperlink);
	g_free (url);

	if (handled)
		return TRUE;

	/* default behavior is to let the terminal widget deal with it */
	if (button_press_event)
		return button_press_event (widget, event);

	return FALSE;
```

It's a bit cleaner in my opinion, and removes all those stray `g_free` calls...